### PR TITLE
Add CI linting for Markdown files

### PR DESCRIPTION
**Pull request by samdbmg**
# Details
Adds linting of Markdown files in the repository using https://github.com/DavidAnson/markdownlint-cli2

Might make it a bit easier to keep the docs neat (and it found a misnumbering for me!). Thoughts?

# Pivotal Story
Story URL: Done as part of https://www.pivotaltracker.com/story/show/186644493

# Related PRs
Follows #15

# Links to external test runs/working deployment
N/A

# Submitter PR Checks
_(tick as appropriate)_

- [X] Added bbc/rd-apmm-cloudfit team as a reviewer
- [ ] PR completes task/fixes bug
- [ ] Tests exercise code appropriately
- [ ] New features and API breaks are flagged in commit messages using magic strings
- [ ] Repo maintainer is notified that a release is required
- [ ] Documentation updated (README, Confluence, Docstrings, API spec, Engineering Guide, etc.)
- [ ] Downstream repos have been checked for potential breaks & fixed as needed
- [ ] APIs/UIs/CLIs updated as required
- [X] PR added to Pivotal story
- [ ] Follow-up stories added to Pivotal
- [ ] Any temporary code/configuration removed (e.g. test deployment environment, temporary commontooling branch)
- [ ] Any pins against pre-releases have been removed

# Reviewer PR Checks
_(tick as appropriate)_

- [ ] PR completes task/fixes bug
- [ ] Tests exercise code appropriately
- [ ] Design makes sense, and fits with our current code base
- [ ] Code is easy to follow
- [ ] PR size is sensible
- [ ] Commit history is sensible and tidy

# Info on Cloudfit PRs
- The checks above are guidelines. They don't all have to be ticked, but they should all have been considered.
- For more details on how to asses PRs, see: https://github.com/bbc/rd-apmm-docs-ways-of-working/blob/master/workflow/PRChecklist.md

### CODE REVIEW

**Comment by samdbmg** [comment]
I've made some fairly substantial changes to how this works (moving from an Action to a Docker image) so I can use the custom one-sentence-per-line rule - requesting re-review as a result

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -30,6 +30,9 @@ jobs:
         with:
           fetch-depth: 0 # check out the entire repo history for SHA verification in lastSuccessful step
 
+      - name: Lint Markdown docs
+        run: make lint-markdown
+
       - name: Lint specification
         run: make lint
```
Comment (j616) [inline .github/workflows/ci-workflow.yml:37]: NIT: Change to `make lint-apispec` in the top dir?

**Review by j616 — APPROVED** [review]
LGTM

**Review by robwadge — APPROVED** [review]
Changes to linting LGTM

File: .github/workflows/ci-workflow.yml (hunk 1 above)
Comment (samdbmg) [reply .github/workflows/ci-workflow.yml]: Good point, changed!

**Pull request merged**